### PR TITLE
Implement rpc remove

### LINKED ISSUE #108: RPC changes
- [ ] Add RPC remove to delete sectors from a contract
- [ ] Remove generic RPC modify

### PR DESCRIPTION
**Pull request by n8maninger**
Closes #108 

note: the failing tests are also in the base branch. Have not dug into them since they look proof related.

### CODE REVIEW

**Comment by lukechampine** [comment]
Anyone feel like bikeshedding the name? `RPCDelete`? `RPCClear`? `RPCFree`? `RPCDeallocate`?

**Comment by n8maninger** [comment]
> Anyone feel like bikeshedding the name? `RPCDelete`? `RPCClear`? `RPCFree`? `RPCDeallocate`?

Delete felt inaccurate to me. Remove feels the most accurate. I don't have a strong preference

File: rhp/v4/server.go [hunk 1]
```diff
@@ -275,53 +256,49 @@ func (s *Server) handleRPCModifySectors(stream net.Conn) error {
...
-		NewMerkleRoot:    rhp4.MetaRoot(roots),
+	roots := state.Roots[:0]
+	for i, root := range state.Roots {
+		if indices[uint64(i)] {
+			continue
+		}
+		roots = append(roots, root)
+	}
+
+	// treeHashes, leafHashes := rhp4.BuildModifySectorsProof(req.Actions, state.Roots) // TODO: build proof
```
Comment (ChrisSchinnerl) [inline rhp/v4/server.go:283]: Are the failing tests related to this?

**Comment by ChrisSchinnerl** [comment]
I don't have a strong preference but out of the listed options I feel like `RPCFree` is probably most accurate. It doesn't delete or remove but essentially frees up space for being reused.

**Comment by lukechampine** [comment]
yeah, I think `RPCFreeSectors` is my preference

File: rhp/v4/server.go (hunk 1 above)
Comment (n8maninger) [reply rhp/v4/server.go:283]: The failing test is in`RPCReadSector`

Fail is here: https://github.com/SiaFoundation/coreutils/blob/ba0a3a3731dbd71e37d946fba1fd7d8fae15afc0/rhp/v4/rpc_test.go#L811

Failing logic here: https://github.com/SiaFoundation/coreutils/blob/ba0a3a3731dbd71e37d946fba1fd7d8fae15afc0/rhp/v4/rpc.go#L228

Comment (n8maninger) [reply rhp/v4/server.go]: Since they're failing on the rhp4 branch as well, I think we can ignore them until the proofs are fixed.

**Review by ChrisSchinnerl — APPROVED** [review]

**Pull request merged**